src/InstallDir/installDirReducer.ts

```diff
@@ -46,65 +46,52 @@ import { RootState } from '../state';
 const CONFIRM_DIR = Symbol('Confirm the install directory');
 const SET_DIR = Symbol('Set the install directory');
 
-const SET_INSTALL_DIR = 'SET_INSTALL_DIR';
-export const setInstallDir = (dir: string) => ({
-    type: SET_INSTALL_DIR,
-    dir,
-});
-
-const SHOW_CONFIRM_INSTALL_DIR_DIALOG = 'SHOW_CONFIRM_INSTALL_DIR_DIALOG';
-export const showConfirmInstallDirDialog = (versionToInstall: string) => ({
-    type: SHOW_CONFIRM_INSTALL_DIR_DIALOG,
-    versionToInstall,
-});
-
-const SHOW_SET_INSTALL_DIR_DIALOG = 'SHOW_SET_INSTALL_DIR_DIALOG';
-export const showSetInstallDirDialog = () => ({
-    type: SHOW_SET_INSTALL_DIR_DIALOG,
-});
-
-const HIDE_INSTALL_DIR_DIALOG = 'HIDE_INSTALL_DIR_DIALOG';
-export const hideInstallDirDialog = () => ({
-    type: HIDE_INSTALL_DIR_DIALOG,
-});
-
-const initialState = () => ({
+type InstallDirectoryState = {
+    currentDir: string;
+    isDialogVisible: boolean;
+    dialogFlavour?: symbol;
+    versionToInstall?: string;
+};
```

Comment:
Nice, with the slice types, we can also replace the current types in `state.ts`.

Reply:
Will do, moving types into reducers.